### PR TITLE
Fix int128 in as_wei_value

### PR DESCRIPTION
**Pull request by TTY-00**
### What I did
move int128 and uint256 into one else branch

### How to verify it
added a test in test_as_wei_value.py

### CODE REVIEW

File: vyper/functions/functions.py [hunk 1]
```diff
@@ -616,7 +616,7 @@ def as_wei_value(expr, args, kwargs, context):
         if value.value <= 0:
             raise InvalidLiteralException("Negative wei value not allowed", expr)
         sub = ['mul', value.value, denom_divisor]
-    elif value.typ.typ == 'uint256':
+    elif value.typ.typ in ['uint256', 'int128']:
```
Comment (charles-cooper) [inline vyper/functions/functions.py]: tuple probably cleaner
```suggestion
    elif value.typ.typ in ('uint256', 'int128'):
```

File: vyper/functions/functions.py [hunk 2]
```diff
@@ -616,7 +616,7 @@ def as_wei_value(expr, args, kwargs, context):
...
         sub = ['mul', value, denom_divisor]
     else:
```
Comment (charles-cooper) [inline vyper/functions/functions.py]: as mentioned in #1637 this should probably be `elif value.typ.typ == 'decimal'`, and then add a catch-all else branch which throws some sort of 'type not found' exception

**Comment by charles-cooper** [comment]
thanks! please see the requested changes and i think this should be good

File: vyper/functions/functions.py (hunk 1 above)
Comment (TTY-00) [reply vyper/functions/functions.py]: Thanks. It's better.

**Comment by robinsierra** [comment]
Can't you get an underflow now if the value is negative?

**Comment by robinsierra** [comment]
If `value` is negative, the result of the multiplication will still be negative. If the result then gets interpreted as a `wei_value` (which I think is unsigned) you get a very large number. Is that correct?

**Comment by charles-cooper** [comment]
so @robinsierra you're saying for `int128` there should be clamp to require a positive value?

**Comment by robinsierra** [comment]
@charles-cooper Yes, I think so.
@fubuloubu For `decimal` as well.

**Comment by TTY-00** [comment]
Since the args may be passed by users, overflow can not be completely avoided at the compiling phase. 
I think the overflow/underflow check may be better to be added in the following two ways, rather than in *as_wei_value*.
1. smart contracts itself, by contract developers.
2. automatically add an overflow/underflow check into the bytecode.

**Comment by fubuloubu** [comment]
Or clamp ge 0 on `value` if not an unsigned type

**Comment by robinsierra** [comment]
The second problem mentioned in #1637 is also still present: for an `uint256` value it is possible to get an overflow, so you need to do the same overflow check as for normal uint256 multiplication.

**Comment by TTY-00** [comment]
That's cool! Thanks for your suggestion.  I think it would be something like this? Or any better way?

```
     ...
    elif value.typ.is_literal:
        if value.value <= 0:
            raise InvalidLiteralException("Negative wei value not allowed", expr)
        
        max_value = int(2**128 / denom_divisor)
        sub = ['mul', ['clample', value.value, max_value], denom_divisor]
    elif value.typ.typ == 'uint256':
        max_value = int(2**256 / denom_divisor)
        sub = ['mul', ['uclample', value, max_value], denom_divisor]
    elif value.typ.typ == 'int128':
        max_value =  int(2**128 / denom_divisor)
        sub = ['mul', ['clample', ['clampge', value, 0], max_value], denom_divisor]
    else:
        max_value = int(2**128 / denom_divisor)
        sub = ['div', ['mul', ['clample', ['clampge', value, 0], max_value], denom_divisor], DECIMAL_DIVISOR]
```
Besides, I think a decimal places check is also needed for *decimal*. But I didn't find a good way.
Do you have any suggestions?

**Comment by charles-cooper** [comment]
I think we should introduce `safe_mul`, `safe_add`, etc to LLL so that we can simplify some of our LLL generation. so then in `expr.py` we just generate `safe_*` ops and then we can reuse the `safe_*` ops in other parts of the codebase like this one. @fubuloubu @jacqueswww thoughts? Or at least create reusable functions for this.

**Comment by fubuloubu** [comment]
probably needs to at least be signed/unsigned versions to, e.g. `safe_add_uint` and `safe_add_int`, or somehow parameterize what underflow/overflow looks like for particular types

**Comment by TTY-00** [comment]
emmm, try it tomorrow

**Comment by charles-cooper** [comment]
@TTY-00 yeah maybe my idea about safe_ ops is not so good - I think there might be too much specialization with basically a different `safe_` op for every single type, like `safe_add_decimal` and so forth. Maybe we can use functions instead (maybe I should do this refactoring in https://github.com/ethereum/vyper/pull/1661)

**Comment by fubuloubu** [comment]
Please address review comments and test failures, then feel free to reopen this PR

**Pull request closed without merging**